### PR TITLE
knn and svm and hyperparameter tuning

### PR DESCRIPTION
**Pull request by KevinLiu-xo**
- created a class for k nearest neighbour and support vector machine (bare bones - legit just inherits everything from base class plus calls the knn or svm api to initate an instance for it)
- also created 2 hyper-parameter tuning methods in the base class
-> grid search (simply tries out all combinations of input parameters to find optimal combination)
-> bayesian search cv one uses some kind of maths model (im not sure about) to learn from previous runs to narrow down and find the optimal hyper-parameters from the input set
-> HELP PART: getting errors trying to pass the range of parameters for hyperparameters into var param space "" TypeError: __init__() got an unexpected keyword argument 'param_space' ""

*this is a snapshot of the error
Screen Shot 2020-09-17 at 3 32 23 pm

-- other then that commented out some code in the assess performance of the base class -> that displays a confusino matrix as a heatmap and the classifcation report i.e. accuracy - its all using libraries like scikit learn btw.   since it seems like something other papers are using confusion matrix to evaluate classification algorithms

### CODE REVIEW

File: model/base_model.py [hunk 1]
```diff
@@ -38,6 +41,37 @@ def __init__(self, data, normalize=False):
...
+
+        # definees the type and number of cross validation splits - refer to: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RepeatedStratifiedKFold.html
+        # Repeated Stratified K Fold -> This repeats a stratified k fold n number of times
+        # Stratified k fold -> Shuffles the data once before splitting into n different parts,
+        # where each part is used as a test set 
+        cv = RepeatedStratifiedKFold(n_splits=5, n_repeats=3, random_state=1)
+
+        if type == 'Grid':
+            # Set all the variables for the grid search cross validation 
+            search = GridSearchCV(estimator=self.model, param_grid=param_space, cv=cv, scoring='accuracy')
```
Comment (ibrahimshaarawy) [inline model/base_model.py:54]: Changed to param_grid as to follow the naming conventions of sklearn

File: model/base_model.py [hunk 2]
```diff
@@ -38,6 +41,37 @@ def __init__(self, data, normalize=False):
...
+
+        elif type == 'Bayesian':
+            # defines the bayes search cv with parameters - refer to: https://scikit-optimize.github.io/stable/modules/generated/skopt.BayesSearchCV.html
+            # Bayesian optimisation is a type of sequential method in which it learns from each step what the optimal hyper-parameters are
+            # (in contrast to grid or random search) - using some complicated maths model (im not sure about)       
+            search = BayesSearchCV(estimator=self.model, param_grid=param_space, n_jobs=-1, cv=cv)
```
Comment (ibrahimshaarawy) [inline model/base_model.py:60]: Changed to param_grid to follow naming conventions of sklearn

**Review by ibrahimshaarawy — COMMENTED** [review]
Changed simple typo, should work now, try testing it first though

**Comment by KevinLiu-xo** [comment]
Cheers Ibrahim :) and yup works

**Pull request merged**